### PR TITLE
Tvw7aqnq: switch to guids from ids

### PR DESCRIPTION
**Pull request by jkunle**
### What
**This part of the  [Pen test remediation card](https://trello.com/c/TvW7aqNq/623-switch-to-guids-from-ids). It aims to replace sequential ids with guids**

-Travis would need to be configured to support postgres.
-Concourse pipeline would need to be adjusted as well.
**Why**
UUIDs or GUIDS are globally unique even across models compared to sequential ids where different models can have the same id. Using GUIDS helps to avoid collisions which can happen with sequential id
From a security perspective using GUIDS makes it substantially more difficult for a malicious entity to guess model id's in the URL
**How**
The approach used was to examine all the tests where ids have been exposed in there raw form and 

1. Refactor where possible with id's generated from model objects i.e.

``` id = 1``` replaced with  ``` id = msa_component.id```

Implement a series of migrations
2. Implement PostgreSQL UUID support into the application using a migration
3. Drop all existing self service tables
4. Create a new migration file from existing **schema.rb** changing id to uuid's in the schema.rb
file.
5. Change database.yml file development and test to use postgresSQL adapters.
SQLite has poor support for GUIDS and potential [workarounds](https://github.com/jashmenn/activeuuid) are currently inadequate.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -29,6 +29,9 @@ class Application < Rails::Application
     # Don't generate system test files.
     config.generators.system_tests = nil
 
+    config.generators do |g|
+      g.orm :active_record, primary_key_type: :uuid
+    end
```
Comment (alan-gds) [inline config/application.rb]: Might be worth having a comment for this, especially given that most of the other stuff in the file has a comment.

File: db/schema.rb [hunk 2]
```diff
@@ -10,65 +10,69 @@
...
     t.string "component_type"
-    t.integer "encryption_certificate_id"
+    t.uuid "encryption_certificate_id"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.boolean "vsp", default: false
-    t.integer "team_id"
-    t.string "environment", null: false
+    t.uuid "team_id"
+    t.string "environment"
```
Comment (alan-gds) [inline db/schema.rb]: How come we're losing `null: false` here?

File: config/application.rb (hunk 1 above)
Comment (jkunle) [reply config/application.rb]: Fair enough, i'll put a comment in.

File: db/schema.rb (hunk 2 above)
Comment (jkunle) [reply db/schema.rb]: Thanks for spotting this; i'll update this

File: config/application.rb (hunk 1 above)
Comment (jkunle) [reply config/application.rb]: I have had to update most of the test due to conflicting changes in master branch

**Pull request merged**